copy/single.go

```diff
@@ -20,6 +20,7 @@ import (
 	compressiontypes "github.com/containers/image/v5/pkg/compression/types"
 	"github.com/containers/image/v5/transports"
 	"github.com/containers/image/v5/types"
+	"github.com/containers/storage/pkg/chunked"
```

Comment:
`pkg/chunked` brings in a _lot_ of dependencies. I’d rather prefer to avoid that in the generic code and non-storage transports.

Reply:
can this be solved with a new package `github.com/containers/storage/pkg/chunked/toc` that includes only the `GetTOCDigest` function?

Reply:
I’d recommend dealing with the way the value is determined, first: Where does the Linux OS dependency decision point live; and how does the generic copy code arrive at the decision to rely on the TOC.

- Hypothetically, if the code detecting this were moved to one of `c/image/{,internal/}{image, manifest}`, I’d strongly prefer for these c/image subpackages not to drag in large dependencies, so a new c/storage subpackage would be best in that case.
- c/image/copy already, via pkg/compression, depends on pkg/chunked/compressor, so if the code detecting this stayed where it is, a new package is not _strictly speaking_ required — the function can be added into …pkg/compressor. As to whether it is a _good fit_ for that package, at this point I don’t have an opinion; it might be better to introduce a new subpackage anyway.
- At the other extreme (my current best guess but didn’t look into it closely), if the code detecting this were moved to c/image/storage, which already depends on all of `c/storage/pkg/chunked`, a new subpackage would not be necessary.

Reply:
`"github.com/containers/storage/pkg/chunked/toc"` was used to isolate the dependency.